Support optional testing of other databases

chosak opened the pull request:
This change adds support for using the `DATABASE_URL` environment variable to specify Django's dev/test database, based on the [dj-database-url](https://github.com/kennethreitz/dj-database-url) package.

It lets you run tests against e.g. Postgres:

```sh
$ DATABASE_URL=postgres://user@localhost/db ./pytest.sh
```

(Note that this will require first creating a local database and `pip install`ing `psycopg2==2.7.3.2`.)

It adds `dj-database-url==0.4.2` to the dev/test requirements as that matches the current version in cfgov-refresh.

It also excludes the Django settings files from coverage because otherwise the new logic would reduce the coverage of this repo.

## Notes

- @richaagarwal and I paired on this. This is part of the work being done on platform#2567 to add Postgres compatibility to all cf.gov code. We're hoping to use this pattern with all satellite apps to make database testing easier everywhere.

## Checklist

* [X] Changes are limited to a single goal (no scope creep)
* [X] Code can be automatically merged (no conflicts)
* [X] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [X] Passes all existing automated tests
* [X] New functions include new tests
* [X] New functions are documented (with a description, list of inputs, and expected output)
* [X] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

chosak committed:
support optional testing of other databases

This change adds support for using the `DATABASE_URL` environment
variable to specify Django's dev/test database, based on the
[dj-database-url](https://github.com/kennethreitz/dj-database-url)
package.

It lets you run tests against e.g. Postgres:

```sh
$ DATABASE_URL=postgres://user@localhost/db ./pytest.sh
```

It adds dj-database-url==0.4.2 to the dev/test requirements as that
matches the current version in cfgov-refresh.

It also excludes the Django settings files from coverage because
otherwise the new logic would reduce the coverage of this repo.

chosak commented:
Travis build seems to have failed for similar reason to #313. @sebworks could this be a Node version thing?

higs4281 reviewed (approved):
very nice! made my day to see tests run against postgres.

higs4281 commented:
> Travis build seems to have failed 

The front-end build (`setup.sh`) uses node 6 and fails with 8.

chosak committed:
enforce use of Node 6 to build the frontend

chosak commented:
Thanks for the review @higs4281!

Interesting, it fails even though the `.travis.yml` file specifies [`nodejs: 6`](https://github.com/cfpb/college-costs/blob/master/.travis.yml#L4).

I've addressed this before in [regulations-site#847](https://github.com/cfpb/regulations-site/pull/847) by explicitly calling `nvm use 6` in the Travis `before_install` step. I added that here in 9d98462 and that seems to have fixed the build.
